app/code/Magento/InventoryConfiguration/Model/StockConfigurationValidator.php

```diff
@@ -0,0 +1,139 @@
+{
+    /**
+     * @var StockItemConfigurationInterface[]
+     */
+    private $validators;
+
+    /**
+     * @var StockItemCriteriaInterfaceFactory
+     */
+    private $stockItemCriteriaFactory;
+
+    /**
+     * @var LegacyStockItemRepository
+     */
+    private $legacyStockItemRepository;
+
+    /**
+     * @var Configuration
+     */
+    private $configuration;
+
+    /**
+     * @var GetProductIdsBySkusInterface
+     */
+    private $getProductIdsBySkus;
+
+    /**
+     * StockConfigurationValidator constructor.
+     * @param array $validators
+     * @param Configuration $configuration
+     * @param LegacyStockItemRepository $legacyStockItemRepository
+     * @param StockItemCriteriaInterfaceFactory $stockItemCriteriaFactory
+     * @param GetProductIdsBySkusInterface $getProductIdsBySkus
+     * @throws LocalizedException
+     */
+    public function __construct(
+        array $validators = [],
+        Configuration $configuration,
+        LegacyStockItemRepository $legacyStockItemRepository,
+        StockItemCriteriaInterfaceFactory $stockItemCriteriaFactory,
+        GetProductIdsBySkusInterface $getProductIdsBySkus
+    ) {
+        $this->configuration = $configuration;
+        $this->legacyStockItemRepository = $legacyStockItemRepository;
+        $this->getProductIdsBySkus = $getProductIdsBySkus;
+        $this->stockItemCriteriaFactory = $stockItemCriteriaFactory;
+
+        foreach ($validators as $validator) {
+            if (!$validator instanceof StockItemConfigurationInterface) {
+                throw new LocalizedException(
+                    __('Validator must implement StockConfigurationInterface.')
+                );
+            }
+        }
+        $this->validators = $validators;
+    }
+
+    /**
+     * Old validation logic
+     *
+     * @param string $sku
+     * @param int $stockId
+     * @param float $qtyWithReservation
+     * @param bool $isSalable
+     * @return bool
+     * @throws LocalizedException
+     */
+    public function execute(string $sku, int $stockId, float $qtyWithReservation, bool $isSalable): bool
+    {
+        //old validation logic below
+        $globalMinQty = $this->configuration->getMinQty();
+        $legacyStockItem = $this->getLegacyStockItem($sku)[0];
+        if (null === $legacyStockItem) {
+            return false;
+        }
+        if ($this->isManageStock($legacyStockItem)) {
+            if (($legacyStockItem->getUseConfigMinQty() == 1 && $qtyWithReservation <= $globalMinQty)
+                || ($legacyStockItem->getUseConfigMinQty() == 0 && $qtyWithReservation <= $legacyStockItem->getMinQty())
+            ) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    /**
+     * @param LegacyStockItem|StockItemInterface $legacyStockItem
+     *
+     * @return bool
+     */
+    private function isManageStock(LegacyStockItem $legacyStockItem): bool
```

Comment:
it's better to have two different class validators 
one responsible for -isManageStock
another for - min quantity